WPFHospitalEditor/Service/MapObjectService.cs

```diff
@@ -38,5 +53,28 @@ private List<MapObject> colorSelectedObject(List<MapObject> allMapObjects)
             }
             return allMapObjects;
         }
+
+        public void checkMapObjectSearchInput(string textBoxInput, string comboBoxTextInput)
+        {
+            HospitalMap.searchResult.Clear();
+            List<MapObject> mapObjects = iMapObjectRepository.getAllMapObjects();
+            if (textBoxInput.Equals("") && comboBoxTextInput.Equals(AllConstants.emptyComboBox)) return;
+            foreach (MapObject mapObject in mapObjects)
+            {
+                if(compareInput(mapObject, textBoxInput, comboBoxTextInput))
+                    HospitalMap.searchResult.Add(mapObject);
+            }
+        }
+
```

Comment:
I think that it is better to rename this function into something like "search(string name, string type)" and function should just return list of MapObjects. Bad thing in this method is existence of side effects (you name it checkMapObjectSearchInput but you actually change hospitalMap.searchResult in it).

Reply:
Fixed.